### PR TITLE
Improve MutationTestingRunner

### PR DESCRIPTION
**Pull request by sanmai**
This PR:

- [x] Changes `MutationTestingRunner` to use a more direct `cast()` instead of `map()`; should save a few cycles on Generator checks inside `map()`
- [x] Adds a TODO note

### CODE REVIEW

File: src/Process/Runner/MutationTestingRunner.php [hunk 1]
```diff
@@ -132,6 +132,7 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
                 return true;
             })
             ->filter(function (Mutant $mutant) {
+                // TODO refactor this comparison into a dedicated comparer to make it possible to swap strategies
```
Comment (sanmai) [inline src/Process/Runner/MutationTestingRunner.php:135]: For example, if a time limit set to 25 seconds, but a whole test suite takes only 20 seconds, we can tell that any set of tests will take less than the timeout. Therefore we can safely skip this check.

Comment (maks-rafalko) [reply src/Process/Runner/MutationTestingRunner.php:135]: > For example, if a time limit set to 25 seconds, but a whole test suite takes only 20 seconds, we can tell that any set of tests will take less than the timeout. Therefore we can safely skip this check.

where we can get this information from, not calling `getNominalTestExecutionTime()`?

**Review by maks-rafalko — APPROVED** [review]

File: src/Process/Runner/MutationTestingRunner.php (hunk 1 above)
Comment (sanmai) [reply src/Process/Runner/MutationTestingRunner.php:135]: From the actual time spent on the initial test run, or from the same junit report. There are multiple opportunities.

**Comment by sanmai** [comment]
Thank you @maks-rafalko

File: src/Process/Runner/MutationTestingRunner.php (hunk 1 above)
Comment (theofidry) [reply src/Process/Runner/MutationTestingRunner.php:135]: 👍  I had some along those lines in mind as well

**Pull request merged**